src/command/backup/file.c

```diff
@@ -285,62 +285,85 @@ backupFile(
                     // Add size filter last to calculate repo size
                     ioFilterGroupAdd(ioReadFilterGroup(storageReadIo(read)), ioSizeNew());
 
-                    // Open the source and destination and copy the file
+                    // Open the source
                     if (ioReadOpen(storageReadIo(read)))
                     {
-                        // Setup the repo file for write. There is no need to write the file atomically (e.g. via a temp file on
-                        // Posix) because checksums are tested on resume after a failed backup. The path does not need to be synced
-                        // for each file because all paths are synced at the end of the backup. It needs to be created in the prior
-                        // context because it will live longer than a single loop when more than one file is being written.
-                        if (write == NULL)
+                        Buffer *const buffer = bufNew(ioBufferSize());
+
+                        do
                         {
-                            MEM_CONTEXT_PRIOR_BEGIN()
+                            ioRead(storageReadIo(read), buffer);
+                            // open destination and copy the file only if read something or not bundling
+                            if (ioReadSize(storageReadIo(read)) > 0 || bundleId == 0)
```

Comment:
Can we use `bufEmpty(buffer)` here instead of `ioReadSize(storageReadIo(read)) > 0`

Reply:
> Can we use `bufEmpty(buffer)` here instead of `ioReadSize(storageReadIo(read)) > 0`

No, because buffer contains headers, which are not null even for zero-size file!

Reply:
So what about moving the size member from IoReadPub to IoReadInterface? Besause it is size of data which are read from IoReadInterface, but not from IoRead.

Reply:
Then we need move local variable `actualBytes` from all functions `THIS_VOID, Buffer *buffer, bool block` to struct `IoReadInterface`

Reply:
moved

Reply:
In my opinion, the patch began to look worse.